src/photos/ducks/timeline/components/Timeline.jsx

```diff
@@ -74,18 +73,32 @@ class Timeline extends Component {
       .downloadArchive(photos.map(({ _id }) => _id), 'selected')
   }
 
-  deletePhotos = (selected, clearSelection) =>
-    confirm(
-      <DeleteConfirm
-        t={this.context.t}
-        count={selected.length}
-        related={belongsToAlbums(selected)}
-      />,
-      () =>
-        Promise.all(selected.map(p => this.props.deletePhoto(p))).then(
-          clearSelection
-        )
-    )
+  closeModal = () => {
+    this.setState({
+      displayModal: false,
+      component: null
+    })
+  }
+
+  deletePhotos = (selected, clearSelection) => {
+    this.setState({
+      displayModal: true,
+      component: (
+        <DeleteConfirm
+          t={this.props.t}
+          count={selected.length}
+          related={belongsToAlbums(selected)}
+          onClose={this.closeModal}
+          confirm={async () => {
+            await Promise.all(
+              selected.map(p => this.props.deletePhoto(p))
+            ).then(clearSelection)
+            this.closeModal()
+          }}
+        />
```

Comment:
For the record, it is a bit tricky to put a component in the state since it will not get the react tree updates. I do not think there is a bug here but it can lead to problems (for ex: https://github.com/cozy/cozy-banks/pull/2042/commits/8ef1ee106c0756d9aeaa4598374cb93b4aa300c0 where we cache an object in the state and then it no longer is updated from the store).

Reply:
Thanks! 5b8218d2c1e791da63e50e8cce6a277a88c5a94c